src/wallet/wallet.cpp

```diff
@@ -2833,6 +2839,9 @@ DBErrors CWallet::LoadWallet(bool& fFirstRunRet)
     if (nLoadWalletRet != DB_LOAD_OK)
         return nLoadWalletRet;
     fFirstRunRet = !vchDefaultKey.IsValid();
+    if (IsWalletFlagSet(WALLET_FLAG_DISABLE_HOT_KEYS)) {
```

Comment:
Why are we touching this logic at all? Seems we should just set a default key for watchonly-only wallets, until default keys are removed normally...